### PR TITLE
add `--num-threads` argument to train cli

### PR DESCRIPTION
**Pull request by mludv**
Hey guys, I did some work on issue #4141 

Take a look and say what you think

**Proposed changes**:
- Adds additional parameters for the NLU training in a similar way to the core training. 
- A `--num-threads` cli arguments is added and forwarded to the  `kwargs` of the NLU components `train` method.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by akelad** [comment]
thanks for submitting this PR, we'll give it a review as soon as possible

File: rasa/train.py [hunk 1]
```diff
@@ -127,10 +134,12 @@ def train(
         train_path: Directory in which to train the model.
         output_path: Output path.
         force_training: If `True` retrain model even if data has not changed.
+        fixed_model_name: Name of model to be stored.
```
Comment (tabergma) [inline rasa/train.py]: You changed the order of arguments, can you please change it back to its original order as this is a public method? Thanks.

**Comment by tabergma** [comment]
@mludv Are you still working on this? If yes, would be great if you could tackle my comments. Would be great if we can merge this soon. Thanks!

**Comment by mludv** [comment]
Hi Tanja,
I got stuck on some tests and some recent changes, I can take a look and hopefully update the PR this weekend

File: rasa/cli/train.py [hunk 2]
```diff
@@ -73,7 +73,8 @@ def train(args: argparse.Namespace) -> Optional[Text]:
         force_training=args.force,
         fixed_model_name=args.fixed_model_name,
         persist_nlu_training_data=args.persist_nlu_data,
-        additional_arguments=extract_additional_arguments(args),
+        core_additional_arguments=extract_core_additional_arguments(args),
+        nlu_additional_arguments=extract_core_additional_arguments(args),
```
Comment (tmbo) [inline rasa/cli/train.py:77]:
```suggestion
        nlu_additional_arguments=extract_nlu_additional_arguments(args),
```

**Pull request merged**